internal/provider/provider.go

```diff
@@ -16,27 +17,62 @@ type Provider struct {
 	version string
 }
 
-// ProviderModel describes the provider data model.
-type ProviderModel struct {
+// Retrieve provider data from configuration
+type providerConfig struct {
+	FulcioURL types.String `tfsdk:"fulcio_url"`
+	RekorURL  types.String `tfsdk:"rekor_url"`
 }
 
 func (p *Provider) Metadata(ctx context.Context, req provider.MetadataRequest, resp *provider.MetadataResponse) {
 	resp.TypeName = "cosign"
 	resp.Version = p.version
 }
 
-func (p *Provider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
+func (p *Provider) Schema(_ context.Context, _ provider.SchemaRequest, resp *provider.SchemaResponse) {
 	resp.Schema = schema.Schema{
-		Attributes: map[string]schema.Attribute{},
+		Attributes: map[string]schema.Attribute{
+			"fulcio_url": schema.StringAttribute{
+				Optional: true,
+			},
+			"rekor_url": schema.StringAttribute{
+				Optional: true,
+			},
```

Comment:
These need defaults

Reply:
I think that should be required too

Reply:
In this struct it does not have the default field, so we default the values in the `Configure` function

Reply:
I'd start by doing this at the resource level instead of the provider level, I missed that this was at the provider when I first looked.

Reply:
I was thinking on that too, ok will move to the resource level

Reply:
I think we should try to get it right on the resources, and then we can support provider-wide configuration of the default next 🤞

Reply:
@mattmoor ok, will make the change later today, thanks for the feedback

Reply:
made in the resource level and update the PR summary, ptal